File: apps/api/FinanceControl.Shared/Dtos/Response/WishlistItemDetailResponseDto.cs
using FinanceControl.Shared.Enums;

namespace FinanceControl.Shared.Dtos.Response
{
    public class WishlistItemDetailResponseDto
    {
        public int Id { get; init; }
        public string Name { get; init; } = string.Empty;
        public string? Description { get; init; }
        public int? TargetPrice { get; init; }
        public EnumWishlistPriority Priority { get; init; }
        public EnumWishlistStatus Status { get; init; }
        public string? Url { get; init; }
        public string? ImageUrl { get; init; }
        public int? LatestPrice { get; init; }
        public DateTime CreatedAt { get; init; }
        public DateTime? UpdatedAt { get; init; }
        public IReadOnlyList<WishlistPriceHistoryItemDto> PriceHistory { get; init; } = [];
    }

    public class WishlistPriceHistoryItemDto
    {
        public int Id { get; init; }
        public int Price { get; init; }
        public DateTime RecordedAt { get; init; }
    }
}

File: apps/mobile/lib/features/wishlist/data/dtos/wishlist_item_detail_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

import 'wishlist_price_history_item_dto.dart';

part 'wishlist_item_detail_response_dto.g.dart';

@JsonSerializable(createToJson: false)
class WishlistItemDetailResponseDto {
  const WishlistItemDetailResponseDto({
    required this.id,
    required this.name,
    this.description,
    this.targetPrice,
    required this.priority,
    required this.status,
    this.url,
    this.imageUrl,
    this.latestPrice,
    required this.createdAt,
    this.updatedAt,
    required this.priceHistory,
  });

  final int id;
  final String name;
  final String? description;
  final int? targetPrice;
  final String priority;
  final String status;
  final String? url;
  final String? imageUrl;
  final int? latestPrice;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final List<WishlistPriceHistoryItemDto> priceHistory;

  factory WishlistItemDetailResponseDto.fromJson(Map<String, dynamic> json) =>
      _$WishlistItemDetailResponseDtoFromJson(json);
}
